app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -314,6 +323,25 @@ public CallInPhoneListener getCallListener() {
         return listener;
     }
 
+    synchronized public Tracker getDefaultTracker() {
+        if (mTracker == null) {
+            // TODO: AMS - Will want to set this conditionally after test release
+            mTracker = getAnalyticsInstance().newTracker(DEV_TRACKING_ID);
+            mTracker.enableAutoActivityTracking(true);
+        }
+        String userId = getCurrentUserId();
+        if (!"".equals(userId)) {
+            mTracker.set("&uid", userId);
+        } else {
+            mTracker.set("&uid", null);
+        }
+        return mTracker;

```

Comment:
`getDefaultTracker` is called every time an event is reported. It is worth thinking about setting up the default tracker in a manner that doesn't require you to set the user id every time `getDefaultTracker` is called. Like if you setup that part of the tracker every time a new user logs in, you don't need to do it every time you call `getDefaultTracker`. That said, it might be a more complex implementation that isn't worth the effort. I'll leave it to you to decide.